flo-workflow/src/main/java/com/spotify/flo/TaskOperator.java

```diff
@@ -39,40 +43,30 @@
     /**
      * Called to report some piece of task metadata.
      *
+     * By default this methods calls {@link #meta(TaskId, Map)} with a single entry.
+     *
      * @param task The task that is being evaluated
      * @param key The metadata key.
      * @param value The metadata value.
      */
-    void meta(TaskId task, String key, String value);
+    default void meta(TaskId task, String key, String value) {
+      meta(task, Collections.singletonMap(key, value));
+    }
 
     /**
      * Called to report some piece of task metadata.
      *
-     * By default this methods calls <code>meta(TaskId task, String key, String value)</code>
-     * for each key-value pair.
-     *
      * @param task The task that is being evaluated
      * @param data The key-value metadata
      */
-    default void meta(TaskId task, Map<String, String> data) {
-      data.forEach((key, value) -> meta(task, key, value));
-    }
+    void meta(TaskId task, Map<String, String> data);
 
-    Listener NOP = (Listener) (task, key, value) -> { };
+    Listener NOP = (task, data) -> { };
 
     default Listener composeWith(Listener listener) {
-      return new Listener() {
-        @Override
-        public void meta(TaskId task, String key, String value) {
-          Listener.this.meta(task, key, value);
-          listener.meta(task, key, value);
-        }
-
-        @Override
-        public void meta(TaskId task, Map<String, String> data) {
-          Listener.this.meta(task, data);
-          listener.meta(task, data);
-        }
+      return (task, data) -> {
+        guardedCall(() -> Listener.this.meta(task, data));
```

Comment:
If a Listener overrides meta(TaskId, String, String), that would not be captured, right? This is a breaking change anyway, so how about we remove that default method?

Reply:
Let's also remove the single-entry for `InstrumentedContext.Listener.meta` method.

Reply:
LGTM. There is not something affecting end user.

Reply:
Done